task-2/src/components/Formik/AddMovie/index.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+
+import { useDispatch } from 'react-redux';
+
+import { Formik, Form } from 'formik';
+import * as Yup from 'yup';
+
+import { submitMovie } from '../../../store/actionCreators';
+import { formikFieldOptions } from '../../../constants';
+
+import { FormikField } from '../FormikField';
+
+import './index.css';
+
+const movieValidationSchema = Yup.object().shape({
+	title: Yup.string().required('Title is required'),
+	release_date: Yup.string().required('Release Date is required'),
+	poster_path: Yup.string().required('Image Url is required'),
+	genres: Yup.string().required('Genre is required'),
+	overview: Yup.string().required('Overview Description is required'),
+	runtime: Yup.number().required('Runtime is required'),
+});
+
+export const AddMovieForm = () => {
+	const [formSubmit, setFormSubmit] = useState(false);
+	const dispatch = useDispatch();
+
+	const submitHandler = (values) => {
+		setFormSubmit(true);
+		dispatch(submitMovie(values));
+	};
+
+	return (
+		<>
+			<Formik
+				initialValues={{
+					title: '',
+					release_date: '',
+					poster_path: '',
+					genres: '',
+					overview: '',
+					runtime: '',
+				}}
+				validationSchema={movieValidationSchema}
+				onSubmit={submitHandler}
+			>
+				{({ errors, touched }) => (
+					<Form>
+						<div>
+							{formikFieldOptions.map((field) => {
```

Comment:
Same rules apply to Formik as well - don't keep JS in JSX, especially mapping. Always try to create separate functions.

Reply:
fixed.